### PR TITLE
Fix memory leak during MHN.resolve

### LINKED ISSUE #11921: OJDK MH: unfreed blocks remaining at shutdown
### Errors
Memory leak issues when OJDK MHs are enabled in OpenJ9.
```
 [ERR] WARNING: 32 unfreed blocks remaining at shutdown!
 [ERR] WARNING: 0 blocks were ignored per ignoredCallsite parameter
 [ERR] port_shutdown_library describing block at 00007FDB900077E0 (header at 00007FDB900075E0):
 [ERR] Wrapped block size is 37, allocation number is 6836
 [ERR] Block was allocated by /home/jenkins/workspace/Build_JDK16_x86-64_linux_ojdk292_Personal/openj9/runtime/vm/stringhelpers.cpp:293
 [ERR] First 32 bytes:
 [ERR] 00007FDB900077E0: 69 6E 76 6F 6B 65 53 74 61 74 69 63 00 90 E7 E7  invokeStatic....
 [ERR] 00007FDB900077F0: E7 E7 E0 77 00 90 E7 E7 E7 E7 E0 77 00 90 E7 E7  ...w.......w....
 [ERR] port_shutdown_library describing block at 00007FDB900080E0 (header at 00007FDB90007EE0):
 [ERR] Wrapped block size is 169, allocation number is 6835
```

### Failing Test Targets

The following test targets are failing with the above error:
1. `cmdLineTester_GCRegressionTests_2` [sanity.functional]
2. `cmdLineTester_GCRegressionTests_3` [sanity.functional]
3. `VmArgumentTests_0` [extended.functional]

### Steps to build an OpenJ9 JDK with OJDK MHs enabled

Refer to #11458.

### PR DESCRIPTION
**Pull request by fengxue-IS**
Fixes memory leak in MHN.resolve if the method lookup failed with exception.

Fixes #11921

### CODE REVIEW

**Comment by fengxue-IS** [comment]
@babsingh can you please take a look

**Review by babsingh — APPROVED** [review]

File: runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp [hunk 1]
```diff
@@ -750,6 +750,8 @@ Java_java_lang_invoke_MethodHandleNatives_resolve(JNIEnv *env, jclass clazz, job
 
 				/* Check for resolution exception */
 				if (VM_VMHelpers::exceptionPending(currentThread)) {
+					j9mem_free_memory(signature);
+					j9mem_free_memory(name);
```
Comment (DanHeidinga) [inline runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp:754]: There are now two places that free these variables - here and line ~245.  If you move the code to free them to after the `done:` label, there will be 1 place making it easier to reason about the allocation lifetimes.  This works becuase it's safe to free `null`

**Comment by babsingh** [comment]
You can also remove:
https://github.com/eclipse/openj9/blob/94eda9a91f1ffacf048628896de4aefaaf7ac7bf/runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp#L705

File: runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp [hunk 2]
```diff
@@ -869,6 +866,9 @@ Java_java_lang_invoke_MethodHandleNatives_resolve(JNIEnv *env, jclass clazz, job
 	}
 
 done:
+	j9mem_free_memory(name);
+	j9mem_free_memory(signature);
```
Comment (DanHeidinga) [inline runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp:870]: Line 705 also does a `j9mem_free_memory(signature)` before a `goto done;` which will double free.

Please check for any other prior free calls covering these variables

**Comment by fengxue-IS** [comment]
Fix double free at line 705. Thanks for pointing out.

**Review by DanHeidinga — APPROVED** [review]

**Comment by DanHeidinga** [comment]
Jenkins test sanity xlinux jdk16

**Comment by babsingh** [comment]
https://github.com/eclipse/openj9/issues/11921 can be closed after this PR is merged.

**Comment by pshipton** [comment]
> #11921 can be closed after this PR is merged.

Added the auto-close keyword to the description.

**Comment by fengxue-IS** [comment]
local testing on JDK16 for tests: `cmdLineTester_GCRegressionTests_2`/`cmdLineTester_GCRegressionTests_3` passed
@DanHeidinga can this be merged?

**Pull request merged**